app/code/Magento/Config/Plugin/App/Config/Type/System/ChangeConfigPath.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Config\Plugin\App\Config\Type\System;
+
+use Magento\Config\App\Config\Type\System;
+use Magento\Framework\App\Config\ConfigPathResolver;
+use Magento\Framework\App\Config\ScopeCodeResolver;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\DeploymentConfig;
+use Magento\Framework\App\ScopeInterface;
+use Magento\Store\Model\ScopeInterface as StoreScopeInterface;
+
+/**
+ * Plugin that for change config path when have deployment config value
+ */
+class ChangeConfigPath
+{
+    /**
+     * @var ScopeCodeResolver
+     */
+    private $scopeCodeResolver;
+
+    /**
+     * Deployment configuration
+     *
+     * @var DeploymentConfig
+     */
+    private $deploymentConfig;
+
+    /**
+     * The resolver for configuration paths according to source type.
+     *
+     * @var ConfigPathResolver
+     */
+    private $configPathResolver;
+
+    /**
+     * @var array
+     */
+    private $websiteCode = [];
+
+    /**
+     * Config constructor.
+     *
+     * @param ScopeCodeResolver $scopeCodeResolver
+     * @param DeploymentConfig $deploymentConfig
+     * @param ConfigPathResolver $configPathResolver
+     */
+    public function __construct(
+        ScopeCodeResolver $scopeCodeResolver,
+        DeploymentConfig $deploymentConfig,
+        ConfigPathResolver $configPathResolver
+    ) {
+        $this->scopeCodeResolver = $scopeCodeResolver;
+        $this->deploymentConfig = $deploymentConfig;
+        $this->configPathResolver = $configPathResolver;
+    }
+
+    /**
+     * Before plugin for change config path base deployment config
+     *
+     * @param System $subject
+     * @param string $path
+     * @return array
+     */
+    public function beforeGet(
+        System $subject,
+        $path = ''
+    ) {
```

Comment:
Please specify the return type 
```suggestion
    ): array {
```